packages/client/screens/UserProfile/UserFollowingSearch/index.tsx

```diff
@@ -0,0 +1,274 @@
+import { UserFollowingSearchScreenProps } from '../../../types';
+
+const ForbiddenAccessToUserFollowingScreen: React.FC<
+    UserFollowingSearchScreenProps & {
+        userProfileInformation: UserProfileInformation;
+    }
+> = ({ navigation, userProfileInformation: { userNickname } }) => {
+    const insets = useSafeAreaInsets();
+
+    return (
+        <AppScreen>
+            <AppScreenHeader
+                title={`${userNickname}'s following`}
+                insetTop={insets.top}
+                canGoBack
+                goBack={() => {
+                    navigation.goBack();
+                }}
+            />
+
+            <AppScreenContainer testID="search-user-following-screen">
+                <Text sx={{ color: 'white', marginBottom: 'xl' }}>
+                    Access to user's following is forbidden
+                </Text>
+
+                <Button title="Go back" onPress={() => navigation.goBack()} />
+            </AppScreenContainer>
+        </AppScreen>
+    );
+};
+
+const UserfollowingScreen: React.FC<UserFollowingSearchScreenProps> = ({
+    navigation,
+    route: {
+        params: { userID: relatedUserID },
+    },
+}) => {
+    const meUSerID = getFakeUserID();
+    const insets = useSafeAreaInsets();
+    const sx = useSx();
+    const [screenOffsetY, setScreenOffsetY] = useState(0);
+    const initialNumberOfItemsToRender = IS_TEST ? Infinity : 10;
+
+    const [userfollowingSearchState, userfollowingSearchMachineSend] =
+        useMachine(
+            createUserFollowingSearchMachine({
+                userID: relatedUserID,
+            }),
+        );
+    const { searchQuery, usersSummaries } = userfollowingSearchState.context;
+    const hasMoreUsersToFetch = userfollowingSearchState.context.hasMore;
+    const isFetching = userfollowingSearchState.hasTag('fetching');
+    const searchBarActor: ActorRef<
+        AppScreenHeaderWithSearchBarMachineEvent,
+        AppScreenHeaderWithSearchBarMachineState
+    > = userfollowingSearchState.children.searchBarMachine;
+    const [searchState, sendToSearch] = useActor(searchBarActor);
+    const showHeader = searchState.hasTag('showHeaderTitle');
+
+    function handleLoadMore() {
+        userfollowingSearchMachineSend({
+            type: 'LOAD_MORE_ITEMS',
+        });
+    }
+
+    function handleRefresh() {
+        userfollowingSearchMachineSend({
+            type: 'REFRESH',
+        });
+    }
+
+    return (
+        <AppScreenWithSearchBar
+            canGoBack
+            goBack={() => {
+                navigation.goBack();
+            }}
+            title="Search for a following"
+            testID="search-user-following-screen"
+            searchInputPlaceholder="Search a following..."
+            showHeader={showHeader}
+            screenOffsetY={showHeader === true ? 0 : screenOffsetY}
+            setScreenOffsetY={setScreenOffsetY}
+            searchQuery={searchQuery}
```

Comment:
<!-- Issues represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 issue:** ‏The search query should be picked from `searchState` and not `userfollowingSearchState.context`. The value of the input must be what was typed previously, and not the query of the last request.

```suggestion
            searchQuery={searchState.context.searchQuery}
```

Reply:
:heavy_check_mark: